src/nupic/math/SparseBinaryMatrix.hpp

```diff
@@ -2002,7 +2002,7 @@ namespace nupic {
 	  << "Invalid index: " << *it
 	  << " - Should be >= 0 and < number of columns:" << nCols();
       
-      for (size_type i = 1; i < (size_type)(end - begin); ++i)
+      for (auto i = 1; i < (end - begin); ++i)
```

Comment:
Actually, and perhaps ironically, I would be fine using `auto` in this cases since it is obviously an iterator type. But given that we are having trouble getting the types right I'd prefer we not use `auto` to be sure that we understand what is happening.